src/puptoo/process/profile.py

```diff
@@ -392,19 +395,30 @@ def system_profile(
             catch_error("uname", e)
             raise
 
-    if redhat_release:
+    if redhat_release and os_release:
         try:
+            # The 'redhat_release.rhel' may be "Red Hat Enterprise Linux xxx"
             profile["os_release"] = redhat_release.rhel
-            profile["operating_system"] = {
-                "major": redhat_release.major,
-                "minor": redhat_release.minor,
-                "name": "RHEL",
-            }
-            if profile.get("host_type") is None:
-                if redhat_release.major >= 8:
-                    profile["system_update_method"] = "dnf"
-                else:
-                    profile["system_update_method"] = "yum" 
+            # But it might not be an RHEL, see
+            # - https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/os_release.py
+            if os_release.is_rhel:
+                profile["operating_system"] = {
+                    "major": redhat_release.major,
```

Comment:
Are these `redhat_release.major` and `redhat_release.minor` not provided for other OS?

Reply:
Yep, for other OS, these two items look pointless, so I leave them as `-1` for NON-RHEL.

Reply:
BTW, just FYI,

According to the `advisor-backend` ([here is the code](https://gitlab.cee.redhat.com/insights-platform/advisor-backend/-/blob/master/service/service.py#L364)), and per my test: When a host is identified as a NON-RHEL via the [OSRelease](https://github.com/RedHatInsights/insights-core/blob/3.0/insights/combiners/os_release.py) combiner, only this [OTHER_LINUX_SYSTEM recommendations](https://console.redhat.com/insights/advisor/recommendations?rule_status=enabled&sort=-total_risk&limit=20&offset=0&text=OTHER_LINUX#SIDs=&tags=) will be shown to the customer on the Advisor UI.

And the update of this PR is trying to make sure the "General Information" ([here](https://console.redhat.com/insights/inventory/72bdbdec-40d6-48ee-934f-e9a80915982b)) on Advisor would be more accurate, even when it's a NON-RHEL.

Reply:
> Yep, for other OS, these two items look pointless, so I leave them as `-1` for NON-RHEL.

I see, in the test examples in [inventory_schema](https://github.com/RedHatInsights/inventory-schemas/blob/master/tests/utils/valids.py#L80), other OS major minor versions are valid. So, should we populate the fields for other OS as well instead of adding `-1`?

Reply:
It's a bit complex..., here in this puptoo `profile` module, the `redhat_release.major/minor` is from the [RedHatReleasse Combiner](https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/redhat_release.py#L38-L51), this combiner tries to get the `major/minor` from the `Uname` parser first which depends on a hardcoded [mapping](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/uname.py#L46-L119)  which is from this [public KCS](https://access.redhat.com/articles/3078), only when the `Uname` parser is not available or [parseable (-1, -1)](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/uname.py#L370), the combiner [turns to check](https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/redhat_release.py#L73) the [RedhatRelease parser](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L35) which just read the literal text of the `etc/redhat-release`.

TLDR;, the `redhat_release.major/minor` is not accurate to indicate the release version, especially when it's other OS.  

For example: assume the host is an `CentOS Linux` but booted with a valid RHEL 7.6 kernel ([valid in this KCS](https://access.redhat.com/articles/3078)), assume its `etc/redhat-release` is `CentOS Linux 7.0`, in this case, the `major.minor` will be set to `7.6` but not `7.0`. the `7.6` is not correct for indicating the release version of CentOS and also not correct for RHEL since it will be identified as `os_release.is_rhel == False`.